### ISSUE TITLE
uuid support without breaking change

### ISSUE THREAD

**Issue by theduke** [sfackler/rust-postgres#359]
Uuid support was already upgraded to 0.6 in master, but not released yet.

It should be perfectly fine to support uuid with a version range of `>= 0.3, < 0.7` since postgres-shared only used `as_bytes` and `from_bytes`, which haven't changed in the last few versions.

This would mean a breaking change is not neccessary.

**Comment by sfackler** [comment]
I'm in the middle of a bunch of changes for the next release, so it might be a bit.

Wide version constraints like `>= 0.3, < 0.7` don't really work very well - if you depend on 0.5, cargo will pick 0.5 for you and 0.6 for postgres.

**Comment by emk** [comment]
As of the writing, it seems like most of the Rust ecosystem is now supporting `uuid` 0.6, which makes it increasingly difficult to get `postgres` working with everything else. I'm not sure what the solution is here.

As a workaround, I'm going to try building a function that uses `as_bytes` and `from_bytes` to pass `uuid::Uuid` values to and from `postgres`.

**Comment by emk** [comment]
It turns out that the easiest workaround right now is to fork `rust-postgres`, update it to use `uuid` 0.6, and then use `[patch]` in `Cargo.toml`:

```
[patch.crates-io]
# Use a patched rust-postgres build supporting uuid 0.6.
postgres = { version = "0.15.2", git = "https://github.com/faradayio/rust-postgres.git", branch = "uuid_0_6" }
postgres-protocol = { version = "0.3.2", git = "https://github.com/faradayio/rust-postgres.git", branch = "uuid_0_6" }
postgres-shared = { version = "0.4.1", git = "https://github.com/faradayio/rust-postgres.git", branch = "uuid_0_6" }
```

**Comment by sfackler** [comment]
In the upcoming release, these features are now versioned, so we can support multiple versions without breaking changes.
